### ISSUE TITLE
Format on save does (sometimes) not work with Pint

### ISSUE THREAD

**Issue by mahoek** [yaegassy/coc-php-cs-fixer#10]
Hi,

I have a very strange issue with the plugin using Pint.

It does not work with format on save but when I run the `:Format` (from the base coc-nvim config) it does the job..

In the `showOutput` it shows that Pint ran and when I debugged the plugin in the class `PintFixFormattingEditProvider` the function `_provideEdits` var `code` has the formatted document.

Been trying to figure out why this is happening but cant solve it.

Also strange is that when I create a new PHP file in my project is does work.

**What i've tried:**

- Disabling all plugins except coc-nvim
- Revert my config and use base coc-nvim from their readme
- Pint from project and downloaded Pint by plugin

**My coc-settings.json:**
```
{
  "php-cs-fixer.activateTool": "pint",
  "coc.preferences.formatOnSaveFiletypes": [
    "php"
  ]
}
```

**Comment by yaegassy** [comment]
Does this problem occur with large files? Perhaps a timeout is occurring. Does changing `coc.preferences.willSaveHandlerTimeout` to a larger value change the behavior?

**coc-settings.json**:

```jsonc
{
  // default: 500 (ms)
  "coc.preferences.willSaveHandlerTimeout": 1000,
}
```

**Comment by mahoek** [comment]
Thanks for the quick reply. It is not a really big file. I tried several times with a higher timeout but doesn't fix it unfortunately. The `:Format` command does it in < 1 sec.

**Comment by yaegassy** [comment]
Thanks for trying. Is there any unusual information output when you run `:CocInfo`?

**Comment by yaegassy** [comment]
Sorry, one correction. it is a `coc.preferences.willSaveHandlerTimeout` setting, but unfortunately, it is not currently respected....

https://github.com/neoclide/coc.nvim/issues/3441

**Comment by mahoek** [comment]
Ah I see, thanks!

Did some tests:
```
./vendor/bin/php-cs-fixer fix   0.27s user 0.05s system 90% cpu 0.355 total
./vendor/bin/pint              0.64s user 0.04s system 97% cpu 0.702 total
```

This explains why php-cs-fixer works and Pint not.

So I guess that is indeed the problem. The file is not that big 650~ lines so maybe this should be something to warn other people (readme?) before they also waste hours debugging this 😂

**Comment by yaegassy** [comment]
I just updated the readme.

By the way, it worked up to about 1700 lines on my machine.
